subsystems/DiscordRPC/src/main/java/org/terasology/engine/subsystem/discordrpc/DiscordRPCThread.java

```diff
@@ -0,0 +1,250 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.subsystem.discordrpc;
+
+import com.jagrosh.discordipc.IPCClient;
+import com.jagrosh.discordipc.IPCListener;
+import com.jagrosh.discordipc.entities.RichPresence;
+import com.jagrosh.discordipc.exceptions.NoDiscordClientException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class DiscordRPCThread implements IPCListener, Runnable {
+    private static final Logger logger = LoggerFactory.getLogger(DiscordRPCThread.class);
+    private static final long DISCORD_APP_CLIENT_ID = 515274721080639504L;
+    private static final String DISCORD_APP_DEFAULT_IMAGE = "ss_6";
+    private static final int MAX_RECONNECT_TRIES = 5;
+
+    private final Thread thread;
+    private final IPCClient ipcClient;
+    private final DiscordRPCBuffer buffer;
+    private RichPresence lastRichPresence;
+
+    private int tries;
+
+    private boolean enabled;
+    private boolean waiting;
+    private boolean connectedBefore;
+    private boolean connected;
+    private boolean autoReconnect;
+
+    public DiscordRPCThread() {
+        thread = new Thread(this);
```

Comment:
I thought of using the `java.util.Timer`. But, that would make handling connection to Discord client harder to do.

Because there are special cases like:
- If the player doesn't have the Discord client we need to try to reconnect then gave up on trying.
- If the player has the Discord client then we connect to it and keep the connection alive. Even if the player disabled the RPC option in the player settings.
- When the player toggles the RPC option in the player settings we just have to send `null` to the Discord client. Instead of closing the connection. And re-connecting again when enabled.